src/asyncOperationManagerUtils.js

```diff
@@ -131,14 +154,32 @@ const getStateForOperationAfterStep = (state, asyncOperationStep, descriptorId,
     otherFields,
   } = getAsyncOperationInfo(newState.descriptors, descriptorId, params);
 
-  // in case operation/descriptor state is initialized in userland we pass that through
-  // to the library state.
+  // descriptor asyncOperationStep callbacks
+  if (asyncOperationDescriptor.onBegin && asyncOperationStep === ASYNC_OPERATION_STEPS.BEGIN_ASYNC_OPERATION) {
```

Comment:
Not that it's really important, but this will never be all three steps at the same time, so it could either be an if/else or a switch (probably nicer syntax).